src/main/java/com/training/elnino/maper/DataRowMapper.java

```diff
@@ -0,0 +1,42 @@
+package com.training.elnino.maper;
+
+import com.training.elnino.model.DataRow;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class DataRowMapper {
+
+    private static final String SPLITTER = " ";
+    private static final String DATE_FORMAT = "yMMdd";
+
+    public DataRow mapToDataRow(String row) {
+        DataRow dataRow = new DataRow();
+
+        String[] data = row.split(SPLITTER);
+        List<String> collect = Stream.of(data).map(s -> s.equals(".") ? "0" : s).collect(Collectors.toList());
+
+        dataRow.setObs(Long.parseLong(collect.get(0)));
+        dataRow.setYear(Integer.parseInt(collect.get(1)));
+        dataRow.setMonth(Integer.parseInt(collect.get(2)));
+        dataRow.setDay(Integer.parseInt(collect.get(3)));
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
```

Comment:
Create a bean, and inject - use prototype or singleton depending on the fact if DateTimeFormatter is thread safe